PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/NsoAdapter.java

```diff
@@ -0,0 +1,179 @@
+        holder.star.setChecked(starredContacts.contains(item.getTitle()));
+        holder.star.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(v.getContext());
+                Set<String> buffer = sharedPref.getStringSet(mContext.getResources().getString(R.string.search_nso_star), new HashSet<String>());
+                Set<String> starredContacts = new HashSet<>(buffer);
+                SharedPreferences.Editor editedPreferences = sharedPref.edit();
+                ToggleButton star = (ToggleButton) v;
+                boolean starred = star.isChecked();
+                String currentTitle = item.getTitle();
+                if (starred) {
+                    if (currentTitle != null) {
+                        starredContacts.add(currentTitle);
+                        editedPreferences.putString(currentTitle + mContext.getResources().getString(R.string.search_nso_star),
+                                currentTitle);
+                    }
+                } else {
+                    starredContacts.remove(currentTitle);
+                    if(currentTitle != null) {
+                        editedPreferences.remove(currentTitle + mContext.getResources().getString(R.string.search_nso_star));
+                    }
+                }
+                editedPreferences.putStringSet(mContext.getResources().getString(R.string.search_nso_star), starredContacts);
+                editedPreferences.apply();
+            }
+        });
+
+        return view;
+    }
+
+    /**
+     * returns the parsed version of the title
+     * @param item the item to be parsed
+     * @return the string to be displayed as the title
+     */
+    public static String getTitleName(RSSItem item) {
+        String title = item.getTitle();
+        title = title.substring(title.indexOf("\">")+2);
+        title = title.substring(0, title.indexOf("</a>"));
+        while (title.contains("&amp;")) {
+            title = title.replace("&amp;", "&");
+        }
+        return title;
+    }
+
+    /**
+     * returns a parsed version of the time of the event
+     * @param item the item to be parsed
+     * @return the string to be displayed as the time
+     * @throws ParseException the exception thrown if couldn't parse correctly.
+     */
+    private String getEventTime(RSSItem item) throws ParseException{
+        String time = item.getTitle();
+        time = time.substring(time.indexOf("event/") + 6);
+        time = time.substring(0, time.indexOf("/"));
+        String starttime = time.substring(0, 17);
+        Calendar cal = Calendar.getInstance();
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd-HHmmss", Locale.US);
+        cal.setTime(sdf.parse(starttime));
+        SimpleDateFormat out = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.US);
```

Comment:
Could we make this more readable? At the very least, seconds are probably irrelevant, but if possible, it'd be nice to have day of the week, and also make this 12-hour time.